lib/tree-view.coffee

```diff
@@ -73,6 +74,14 @@ class TreeView
     @element.style.width = "#{state.width}px" if state.width > 0
     @attach() if state.attached
 
+    @disposables.add @onEntryMoved ({initialPath, newPath}) ->
+      updateEditorsForPath(initialPath, newPath)
+
+    @disposables.add @onEntryDeleted ({path}) ->
+      for editor in atom.workspace.getTextEditors()
+        if editor?.getPath()?.startsWith(path)
+          editor.destroy()
```

Comment:
Hm, this code path didn't change (it was only moved). I'll double check again but I was pretty sure that it left the editor open with unsaved changes.